src/Table.tsx

```diff
@@ -71,14 +73,16 @@ export function Table<RowShape>({
       </thead>
       <tbody>
         {data.map((item, index) => (
-          <tr>
+          <tr key={index}>
```

Comment:
Ah good idea adding keys, we do need this  

We shouldn't use the positional index as the `key` though (this is actually the same as not specifying the prop at all)  

Say we have 3 rows (A, B, C), and we were to delete A to result in 2 rows of (B, C), we'll end up rendering (A, B) because according to the keys, `3` was removed, `1` and `2` haven't changed. ([a longer article on this](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318))

One solution would be to have `<Table/>` accept a `keyOn` prop (analogous to react-table's `indexKey` prop), then we can `<tr key={item[props.keyOn]}/>`

Reply:
@cheapsteak typescript wont allow us to access item[props.keyOn] unless keyOn is a single type of RowShape. I don't know how to type something as "one of the fields of a type", I know Partial exists but you can't access an element on a Partial typed key. Thoughts?

Reply:
How's this? 

```diff
@@ -36,14 +36,18 @@ interface Props<RowShape> {
       list: ReadonlyArray<RowShape>
     ) => React.ReactNode;
   }>;
+
+  keyOn: string | ((row: RowShape) => string | number);
 }
 
 export function Table<RowShape>({
   data,
   density = "standard",
   columns,
+  keyOn,
 }: Props<RowShape>): ReturnType<React.FC> {
   const padding = density === "standard" ? 8 : density === "condensed" ? 3 : 11;
+  const getRowKey = typeof keyOn === 'function' ? keyOn : (row) => row[keyOn];
 
   return (
     <table
@@ -77,7 +81,7 @@ export function Table<RowShape>({
       </thead>
       <tbody>
         {data.map((item, index) => (
-          <tr key={index}>
+          <tr key={getRowKey(item)}>
             {columns.map(
               ({
                 render,

```

(dont see any ts errors in my editor at least)

Reply:
oo nice! thanks

Reply:
My advice would be to go even more extreme: provide a function to get the key and if that function doesn't exist, then use `JSON.stringify` on the whole row.